build.sbt

```diff
@@ -26,6 +26,7 @@ lazy val jawnSettingsJS = List(
     _.withSemantics(
       _.withAsInstanceOfs(org.scalajs.linker.interface.CheckedBehavior.Unchecked)
         .withArrayIndexOutOfBounds(org.scalajs.linker.interface.CheckedBehavior.Unchecked)
+        .withStringIndexOutOfBounds(org.scalajs.linker.interface.CheckedBehavior.Unchecked)
```

Comment:
`.withStringIndexOutOfBounds` was introduced in `1.11.0` (see https://www.scala-js.org/news/2022/09/15/announcing-scalajs-1.11.0) and just like with the other checked exception settings we need to set it to unchecked behaviour since Scala.js relies on it for performance.

Reply:
Seb and I had a [detailed Discord discussion](https://discord.com/channels/632150470000902164/635668814956068864/1110694649770819766) about this issue.

Although I originally introduced these semantics modifications, it's actually not the right thing to do. Instead of doubling-down on this wrong behavior, we should fix the root cause, especially now that Scala.js 1.13 has even higher fidelity for catching these sorts of issues. We should fix this before the next release, in case it causes trouble for downstreams.

Reply:
Thanks I just read the conversation, just ping me when your PR fix gets merged and I will then rebase this PR.

Reply:
Resolved via https://github.com/typelevel/jawn/pull/537